swift/ql/src/queries/Security/CWE-916/InsufficientHashIterations.ql

```diff
@@ -0,0 +1,69 @@
+/**
+ * @name Insufficient hash iterations
+ * @description Using hash functions with < 1000 iterations is not secure, because that scheme leads to password cracking attacks due to having an insufficient level of computational effort.
+ * @kind path-problem
+ * @problem.severity error
+ * @security-severity 7.8
+ * @precision high
+ * @id swift/insufficient-hash-iterations
+ * @tags security
+ *       external/cwe/cwe-916
+ */
+
+import swift
+import codeql.swift.dataflow.DataFlow
+import codeql.swift.dataflow.TaintTracking
+import DataFlow::PathGraph
+
+/**
+ * An `Expr` that is used to initialize a password-based encryption key.
+ */
+abstract class IterationsSource extends Expr { }
+
+/**
+ * A literal integer that is 1000 or less is a source of taint for iterations.
+ */
+class IntLiteralSource extends IterationsSource instanceof IntegerLiteralExpr {
+  IntLiteralSource() { this.getStringValue().toInt() < 1000 }
+}
+
+/**
+ * A class for all ways to set the iterations of hash function.
+ */
+class InsufficientHashIterationsSink extends Expr {
+  InsufficientHashIterationsSink() {
+    // `iterations` arg in `init` is a sink
+    exists(ClassOrStructDecl c, AbstractFunctionDecl f, CallExpr call, int arg |
+      c.getFullName() = ["PBKDF1", "PBKDF2"] and
+      c.getAMember() = f and
+      f.getName().matches("init(%iterations:%") and
```

Comment:
This is slightly different from how we've usually matched function names before (listing all the variations explicitly).  I think I'm happy with it, it will be more robust if new versions of `init` with slightly different parameters are added in future.

Reply:
sgtm!